From bd6bc7ebb48adad04a6a115f6f4167ae40d9f697 Mon Sep 17 00:00:00 2001
Date: Tue, 7 Mar 2017 18:28:42 -0800
Subject: [PATCH] Define O_FSYNC in libconstants.gen.c

Musl libc prior to c1f4ed150137d793c9d07356305a89e8785e7e02 has a bug
in which it uses O_FSYNC without defining it anywhere.  Work around it
by explicitly defining O_FSYNC as O_SYNC as musl does with that
commit.
---
 gen_constants.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gen_constants.sh b/gen_constants.sh
index ed0956e..13bc00f 100755
--- a/gen_constants.sh
+++ b/gen_constants.sh
@@ -48,6 +48,11 @@ SED_MULTILINE='s@#define ([[:upper:]][[:upper:]0-9_]*).*@#ifdef \1\
 # macro functions that don't evaluate to a constant.
 cat <<-EOF > "${OUTFILE}"
 /* GENERATED BY MAKEFILE */
+
+#ifndef O_FSYNC
+#define O_FSYNC O_SYNC
+#endif
+
 #include "gen_constants-inl.h"
 #include "libconstants.h"
 const struct constant_entry constant_table[] = {
--
2.16.2
